src/tests/example.spec.js

```diff
@@ -28,3 +27,154 @@ test.describe('Saucedemo app basic tests', () => {
         await expect(app.shoppingCart.cartItems).not.toBeAttached();
     });
 });
+
+
+
+test.describe('Saucedemo UNIT 10', () => {
+        test.beforeEach( async (/** @type {{ app: import('../pages/Application').Application }} */{ app },) =>{
+            await app.login.navigate();
+            await app.login.performLogin('standard_user', 'secret_sauce');
+        } );
+
+
+        test('1. Check sorting by Name A Z', async ({app})=> {
+                  await app.inventory.sortField.selectOption('az');
+                  const goodsName = await app.inventory.inventoryItemsName.allTextContents();
+                  await expect([...goodsName].sort()).toEqual(goodsName);
+          });
+
+        test('1. Check sorting by Name Z A', async ({app})=> {
+                await app.inventory.sortField.selectOption('za');
+                const goodsName = await app.inventory.inventoryItemsName.allTextContents();
+ //             await expect([...goodsName].sort().reverse()).toEqual(goodsName);
+                await expect([...goodsName].sort((a, b) => b.localeCompare(a))).toEqual(goodsName);
+        });
+
+        test('1. Check sorting by Price Low to high', async ({app}) => {
+                await app.inventory.sortField.selectOption('lohi');
+                const goodsPriceText = await app.inventory.inventoryItemsPrice.allTextContents();
+                const goodsPrice = goodsPriceText.map(price => parseFloat(price.replace('$', '')));     
+                await expect([...goodsPrice].sort((a, b) => a - b)).toEqual(goodsPrice);
+        });
+
+        test('1. Check sorting by Price High to Low', async ({app}) => {
+                await app.inventory.sortField.selectOption('hilo');
+                const goodsPriceText = await app.inventory.inventoryItemsPrice.allTextContents();
+                const goodsPrice = goodsPriceText.map(price => parseFloat(price.replace('$', '')));     
+                await expect([...goodsPrice].sort((a, b) => b - a)).toEqual(goodsPrice);
+        });
+
+
+
+        test('test2 Add several random products to the Shopping Cart ana Verify  Name, Description, Price', async ({app}) =>{
```

Comment:
Its hard to read this code because of the spacing. Please follow the best practices for proper spacing here, or use prettier